### ISSUE TITLE
[rllib] Support multiple instances of training within the same process

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#860]
Currently, due to variable conflicts you cannot instantiate multiple instances of algorithms within the same process. In addition, attempts to save/restore checkpoints will restore variables of all algorithms.

To fix this each algorithm should create its graph within a unique `tf.variable_scope`.

Example: we shouldbe be able to support some workflow like the following:
```
import ray

from ray.rllib.policy_gradient import PolicyGradient, DEFAULT_CONFIG

ray.init()
config = DEFAULT_CONFIG.copy()
config["num_sgd_iter"] = 5
config["timesteps_per_batch"] = 1000
alg1 = PolicyGradient('CartPole-v0', config)
alg2 = PolicyGradient('CartPole-v0', config)

res = alg1.train()
while res.episode_reward_mean < 100:
    res = alg1.train()

alg2.restore(res.checkpoint_path)
while res.episode_reward_mean < 190:
    res = alg2.train()
```

**Comment by robertnishihara** [comment]
How does that solution compare with just creating each model within a new graph? E.g.,

```python
with tf.Graph().as_default():
    ...
```

**Comment by ericl** [comment]
Yeah that's probably better.

**Comment by robertnishihara** [comment]
Fixed by #875.
